src/v/ssx/metrics.h

```diff
@@ -81,18 +83,105 @@ class metric_groups {
     // Add metrics belonging to the same group.
     //
     // This is the same method as above but using an explicit vector.
-    metric_groups& add_group(
+    metric_groups_base& add_group(
       const ss::metrics::group_name_type& name,
       const std::vector<ss::metrics::metric_definition>& l) {
         _underlying.add_group(name, l);
         return *this;
     }
 
-    // Clear all metric registrations.
-    void clear() { _underlying.clear(); }
+protected:
+    ss::metrics::metric_groups _underlying;
+};
+
+class internal_metric_groups : public metric_groups_base {
```

Comment:
Missing class comment.

Reply:
Added (as well as to public_metrics_group).